webapp/components/test-runs-query-builder.html

```diff
@@ -239,25 +243,29 @@ <h3>
       // Respond to shas URL changing by fetching the shas
       shasURLUpdated(url) {
         fetch(url).then(r => r.json()).then(s => {
-          this.shas = s;
+          this.matchingSHAs = s;
         });
       }
 
       // Respond to newly fetched shas, or user input, by filtering the autocomplete list.
-      shasUpdated(sha, shas) {
-        if (!shas || !shas.length || !this.queryBuilderSHA) {
+      shasUpdated(sha, matchingSHAs) {
+        if (!matchingSHAs || !matchingSHAs.length || !this.queryBuilderSHA) {
           return;
         }
         if (sha) {
-          shas = shas.filter(s => s.startsWith(sha));
+          matchingSHAs = matchingSHAs.filter(s => s.startsWith(sha));
```

Comment:
Are matching shad guaranteed to be 40-char shas? If not the reverse condition seems like it's also needed. If so, maybe a helper to compare shas?

Reply:
No; they're 10-char SHAs at the moment. This function is basically filtering a complete list of 10-char shas into a filtered list based on what you've typed (prefix matching). I'll be cleaning up the /api/shas endpoint to accept a length param at some point, and we can revise this then; however, nobody would type the 40 char sha. They'd paste it, and having the value not match autocomplete won't be an issue.